src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -141,3 +149,36 @@ def resolve_root_node_address(nodes: str) -> str:
         nodes = re.sub(r"\[(.*?)[,-].*\]", "\\1", nodes)  # Take the first node of every node range
         nodes = re.sub(r"\[(.*?)\]", "\\1", nodes)  # handle special case where node range is single number
         return nodes.split(" ")[0].split(",")[0]
+
+    @staticmethod
+    def _validate_srun_used() -> None:
+        """Checks if the `srun` command is available and used.
+
+        Parallel jobs (multi-GPU, multi-node) in SLURM are launched by prepending `srun` in front of the Python command.
+        Not doing so will result in processes hanging, which is a frequent user error. Lightning will emit a warning if
+        `srun` is found but not used.
+        """
+        srun_exists = subprocess.call(["which", "srun"]) == 0
+        srun_used = "SLURM_NTASKS" in os.environ
```

Comment:
```suggestion
        srun_used = SLURMEnvironment.detect()
```

when not calling in `detect` (see my comment above), we could reuse this in case the implementation might change :)

Reply:
This would result in infinite recursion, since we need to check it in detect() itself too.

Reply:
srun_used could become an utility fn in this file to reuse the implementation, where  both detect and this one call it.

Reply:
I did this, although I find it adds an indirection that makes the code less readable.